src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -711,5 +712,63 @@ static void Main(string[] args)
 
             VisualStudio.Editor.Verify.CodeActions(expectedItems, ensureExpectedItemsAreOrdered: true);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsConfiguration)]
+        public void ConfigureSeverity()
+        {
+            var markup = @"
+class C
+{
+    public static void Main()
+    {
+        // CS0168: The variable 'x' is declared but never used
+        int $$x;
+    }
+}";
+            SetUpEditor(markup);
+
+            // Verify CS0168 warning in original code.
+            VerifyDiagnosticInErrorList("Warning", VisualStudio);
+
+            // Apply configuration severity fix to change CS0168 to be an error.
+            SetUpEditor(markup);
+            VisualStudio.Editor.InvokeCodeActionList();
+            var expectedItems = new[]
+            {
+                "Remove unused variable",
+                "Suppress or Configure issues",
+                    "Suppress CS0168",
+                        "in Source",
+                    "Configure CS0168 severity",
+                        "None",
+                        "Silent",
+                        "Suggestion",
+                        "Warning",
+                        "Error",
+            };
+            VisualStudio.Editor.Verify.CodeActions(expectedItems, applyFix: "Error", ensureExpectedItemsAreOrdered: true);
```

Comment:
Is there some way to have this validate the preview?

Reply:
I tried to thread in a delay between the action being computed and applied, and the integration test does wait after bringing up the light bulb, but the integration test does not expand the nested actions and hence do not bring up the preview. We either need to figure out some other way to test preview OR write a dummy test analyzer that registers a top level code action to update the .editorconfig, though I am not sure if our integration tests can run with a custom defined analyzer that is not in the box.

In short, I could not find an easy way to validate the preview.